### PR TITLE
Top Posts Widget: remove limitation preventing the number of days being extended

### LINKED ISSUE #3977: Top posts limited to 10 days
I expected the top posts to be able to pull in data from the past week, month and year.

But instead there is a block in place which prevents it going past 10 days
https://github.com/Automattic/jetpack/blob/master/modules/widgets/top-posts.php#L467

Interestingly, the API is able to access the extended data. I tested this by simply removing line 467 from the modules/widgets/top-posts.php file.

If there is no reason for this to be the case, then I'll happily submit a pull request to remove this limitation or a filter to modify the maximum number of days.

### PR DESCRIPTION
**Pull request by ryanhellyer**
Fixes #3977 .

I expected the top posts to be able to pull in data from the past week, month and year. But instead, there is a limitation in place which prevents it going past 10 days or below two days, although the API is able to access this extra data.
https://github.com/Automattic/jetpack/blob/master/modules/widgets/top-posts.php#L467

#### Changes proposed in this Pull Request:
Remove the limtitation prevent the number of days being extended.
Add additional inline documentation explaining when and how to use caching to prevent excessively long queries from being run frequently.
Add additional inline documentation explaining that -1 gives access to an infinite number of days.

### CODE REVIEW

File: modules/widgets/top-posts.php [hunk 1]
```diff
@@ -445,6 +445,9 @@ function get_by_views( $count, $args ) {
 
 		/**
 		 * Filter the number of days used to calculate Top Posts for the Top Posts widget.
+		 * We do not recommend accessing more than 10 days of results at one.
```
Comment (jeherve) [inline modules/widgets/top-posts.php]: Could you add an empty line between the filter title and the description? Our parser likes that better, when outputting documentation.

File: modules/widgets/top-posts.php [hunk 2]
```diff
@@ -455,14 +458,6 @@ function get_by_views( $count, $args ) {
 		 */
 		$days = (int) apply_filters( 'jetpack_top_posts_days', 2, $args );
 
-		if ( $days < 1 ) {
```
Comment (jeherve) [inline modules/widgets/top-posts.php]: This was originally added to handle cases when `$days` is zero or evaluates to `false` after passing it through the filter and being cast to an `int`.

We can't keep it as is if we want to be able to handle `-1`, but it'd be nice to keep something that would default to `2` when `$days` is zero or `false`.

Comment (ryanhellyer) [inline modules/widgets/top-posts.php]: Sorry for the delay in responding. The last commit I added deals with that scenario.

**Comment by defries** [comment]
Any news on this? :)

**Comment by jeherve** [comment]
@defries The change is planned for 4.1, currently scheduled for release on June 28, 2016. We'll make sure to include that PR in the 4.1 Beta 👍

**Comment by samhotchkiss** [comment]
LGTM 👍

**Pull request merged**